gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/ComposePlugin.kt

```diff
@@ -69,89 +65,120 @@ abstract class ComposePlugin : Plugin<Project> {
         val material3 get() = composeMaterial3Dependency("org.jetbrains.compose.material3:material3")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.material3:material3-adaptive-navigation-suite:${ComposeBuildConfig.composeMaterial3Version}\""))
         val material3AdaptiveNavigationSuite get() = composeMaterial3Dependency("org.jetbrains.compose.material3:material3-adaptive-navigation-suite")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.runtime:runtime:${ComposeBuildConfig.composeVersion}\""))
         val runtime get() = composeDependency("org.jetbrains.compose.runtime:runtime")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.runtime:runtime-saveable:${ComposeBuildConfig.composeVersion}\""))
         val runtimeSaveable get() = composeDependency("org.jetbrains.compose.runtime:runtime-saveable")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.ui:ui:${ComposeBuildConfig.composeVersion}\""))
         val ui get() = composeDependency("org.jetbrains.compose.ui:ui")
-        @Deprecated("Use desktop.uiTestJUnit4", replaceWith = ReplaceWith("desktop.uiTestJUnit4"))
-        @ExperimentalComposeLibrary
-        val uiTestJUnit4 get() = composeDependency("org.jetbrains.compose.ui:ui-test-junit4")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.ui:ui-test:${ComposeBuildConfig.composeVersion}\""))
         @ExperimentalComposeLibrary
         val uiTest get() = composeDependency("org.jetbrains.compose.ui:ui-test")
+        @Deprecated("Use org.jetbrains.compose.ui:ui-tooling-preview module instead", replaceWith = ReplaceWith("\"org.jetbrains.compose.ui:ui-tooling:${ComposeBuildConfig.composeVersion}\""))
         val uiTooling get() = composeDependency("org.jetbrains.compose.ui:ui-tooling")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.ui:ui-util:${ComposeBuildConfig.composeVersion}\""))
         val uiUtil get() = composeDependency("org.jetbrains.compose.ui:ui-util")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.ui:ui-tooling-preview:${ComposeBuildConfig.composeVersion}\""))
         val preview get() = composeDependency("org.jetbrains.compose.ui:ui-tooling-preview")
+        @Deprecated("Specify dependency via version catalog", replaceWith = ReplaceWith("\"org.jetbrains.compose.material:material-icons-extended:1.7.3\""))
         val materialIconsExtended get() = "org.jetbrains.compose.material:material-icons-extended:1.7.3"
         val components get() = CommonComponentsDependencies
```

Comment:
Do we need a deprecation for `components`?
I think it's similar to `html` below.